spec/AMQP/21-AMQP_FORMAT_RATIONALE.md

```diff
@@ -0,0 +1,86 @@
+# Rationale for AMQP protocol decisions
+
+## Strings vs. complex types
+
+Complex types in most cases cheaper than string representation of `traceparent`
+and `tracestate`. It is important in size restricted environments like IoT and
+price-sensitive infrastructures where one pays for message metadata stored in a
+queue server for a long time.
+
+Complex types are well defined and widely used in AMQP and they are part of a
+protocol. So using them is fully supported by all existing clients.
+
+## Binary `traceparent` vs. list of binary values
+
+There are multiple ways to implement a `traceparent`. It can either be
+implemented as string (http-like), binary protocol (like for grpc) or list of
+separate binary values (`trace-id`, `parent-id`, `trace-flags`).
+
+Strings duplicating the size of a field, using list of binaries will require to
+redefine the way the field serialized, parsed, and versioned. So re-using binary
+protocol looks like a logical solution.
```

Comment:
Where is that 'binary protocol' defined? (I found https://w3c.github.io/trace-context/extension-binary.html but it seems to be empty?). AMQP has quite a complex type-system that will already be required to decode the message, so it might be simpler to use that to encode traceparent in some way rather than requiring further decoding of the field.

Reply:
It is in another PR: https://github.com/w3c/trace-context/pull/215 Sorry for confusion.

@grs what type would you suggest for the header? We discussed options and didn't find the right match

Reply:
Actually, to @calohmn's point re the restrictions on application-property values, perhaps you are right and the binary format is best (it is not very arduous to parse). (Alternatively an amqp list or a map with integer keys would be similar to the binary encoding).

Reply:
@grs yeah. It may be better to switch to binary protocol than. At least some consistency - either text (like HTTP or MQTT) or binary (like GRPC or AMQP).